packages/editor/src/index.ts

```diff
@@ -26,34 +59,21 @@ export default createPrompt<string, EditorConfig>((config, done) => {
     }
 
     if (isEnterKey(key)) {
-      rl.pause();
-      editAsync(
-        value,
-        async (error, answer) => {
-          rl.resume();
-          if (error) {
-            setError(error.toString());
-          } else {
-            setStatus('loading');
-            const isValid = await config.validate(answer);
-            if (isValid === true) {
-              setError(undefined);
-              setStatus('done');
-              done(answer);
-            } else {
-              setValue(answer);
-              setError(isValid || 'You must provide a valid value');
-              setStatus('pending');
-            }
-          }
-        },
-        {
-          postfix: config.postfix || '.txt',
-        }
-      );
+      startExternalEditor(rl);
     }
   });
 
+  if (config.waitUserInput === false && firstRender) {
+    useEffect(() => {
```

Comment:
By React hooks rules, they must be invoked outside conditional structure. So the `if` should be moved inside the `useEffect` callback.